lib/update_signing_flow.py

```diff
@@ -43,3 +46,33 @@ def update_signing_flow(signflow_uri: str):
                                                        signflow_uri,
                                                        doc["uri"])
         agent_update(wrap_up_qs)
+
+def sync_signers_status(sig_flow, sh_workflow_details):
+    sh_workflow_users = sh_workflow_details["users"]
+    kaleidos_signers = get_signers(sig_flow, agent_query)
+    logger.debug(f"Syncing signers status ...")
+    for sh_workflow_user in sh_workflow_users:
+        proc_stat = sh_workflow_user["process_status"]
+        logger.debug(f"Signer {sh_workflow_user['user_email']} has process status {proc_stat}.")
+        kaleidos_signer = next(filter(lambda s: s["email"] == sh_workflow_user["user_email"], kaleidos_signers), None)
+        if kaleidos_signer["end_date"]:
+            logger.info(f"Signer {kaleidos_signer['email']} already has an end date in our db. No syncing needed.")
+        if kaleidos_signer and not kaleidos_signer["end_date"]:
+            if proc_stat == "UN_PROCESSED":
+                pass
+            # elif proc_stat == "IN_PROGRESS":
+            elif proc_stat == "SIGNED": # "REVIEWED",
+                logger.info(f"Signer {kaleidos_signer['email']} signed. Syncing ...")
+                signing_time = sh_workflow_user["processed_on"]
+                if signing_time != kaleidos_signer["end_date"]:
+                    query_string = construct_update_signing_activity_end_date(
+                        sig_flow,
+                        kaleidos_signer["uri"],
+                        datetime.fromisoformat(kaleidos_signer["end_date"]))
```

Comment:
Should be `signing_time` instead of `kaleidos_signer["end_date"]

Reply:
check!